Gem/Code/Source/SampleComponentManager.cpp

```diff
@@ -486,6 +485,20 @@ namespace AtomSampleViewer
             }
             AZ_Warning("SampleComponentManager", targetSampleFound, "Failed find target sample %s", targetSampleName.c_str());
         }
+        if (commandLine->HasSwitch("timingSamples"))
+        {
+                AZStd::string timingSamplesStr = commandLine->GetSwitchValue("timingSamples", 0);
+                int timingSamplesCount = atoi(timingSamplesStr.c_str());
+                if (timingSamplesCount <= 0 || timingSamplesCount > 1000000)
+                {
+                    timingSamplesCount = static_cast<int>(FrameTimeLogSize);
+                }
+                m_imGuiFrameTimer = AZStd::make_unique<ImGuiHistogramQueue>(timingSamplesCount, timingSamplesCount, 250.0f);
+        }
+        else
+        {
+            m_imGuiFrameTimer = AZStd::make_unique<ImGuiHistogramQueue>(FrameTimeLogSize, FrameTimeLogSize, 250.0f);
```

Comment:
Can this still be created on construction and replaced if the command line option is specified and valid?

Reply:
done